app/backend/approaches/chatreadretrieveread.py

```diff
@@ -96,106 +96,89 @@ async def run_until_final_call(
-
-        # STEP 2: Retrieve relevant documents from the search index with the GPT optimized query
-
-        # If retrieval mode includes vectors, compute an embedding for the query
-        vectors: list[VectorQuery] = []
-        if use_vector_search:
-            vectors.append(await self.compute_text_embedding(query_text))
-
-        results = await self.search(
-            top,
-            query_text,
-            filter,
-            vectors,
-            use_text_search,
-            use_vector_search,
-            use_semantic_ranker,
-            use_semantic_captions,
-            minimum_search_score,
-            minimum_reranker_score,
-        )
-
-        sources_content = self.get_sources_content(results, use_semantic_captions, use_image_citation=False)
-        content = "\n".join(sources_content)
+            query_text = self.get_search_query(chat_completion, original_user_query)
 
-        # STEP 3: Generate a contextual and content specific answer using the search results and chat history
+            # STEP 2: Retrieve relevant documents from the search index with the GPT optimized query
 
-        # Allow client to replace the entire prompt, or to inject into the exiting prompt using >>>
-        system_message = self.get_system_prompt(
-            overrides.get("prompt_template"),
-            self.follow_up_questions_prompt_content if overrides.get("suggest_followup_questions") else "",
-        )
+            # If retrieval mode includes vectors, compute an embedding for the query
+            vectors: list[VectorQuery] = []
+            if use_vector_search:
+                vectors.append(await self.compute_text_embedding(query_text))
 
-        response_token_limit = 1024
-        messages = build_messages(
-            model=self.chatgpt_model,
-            system_prompt=system_message,
-            past_messages=messages[:-1],
-            # Model does not handle lengthy system messages well. Moving sources to latest user conversation to solve follow up questions prompt.
-            new_user_content=original_user_query + "\n\nSources:\n" + content,
-            max_tokens=self.chatgpt_token_limit - response_token_limit,
-            fallback_to_default=self.ALLOW_NON_GPT_MODELS,
-        )
+            results = await self.search(
+                top,
+                query_text,
+                filter,
+                vectors,
+                use_text_search,
+                use_vector_search,
+                use_semantic_ranker,
+                use_semantic_captions,
+                minimum_search_score,
+                minimum_reranker_score,
+            )
 
-        data_points = {"text": sources_content}
+            sources_content = self.get_sources_content(results, use_semantic_captions, use_image_citation=False)
+            if sources_content:
+                extra_info["data_points"] = {"text": sources_content}
 
-        extra_info = {
-            "data_points": data_points,
```

Comment:
You seem to have lost the data_points from thought process, please bring them back.

Reply:
This behavior is intentional, as there are no data points when RAG is disabled. However, the solution could be more elegant than simply overwriting the `extra_info` variable.